### DISCUSSION TITLE
How to filter a matrix with an index？

### DISCUSSION THREAD

**Discussion by wujilingfeng** [AccelerateHS/accelerate#519] [Q&A] [answered]
I want to filter a matrix with index info and record it's index for other use, so how to filter a matrix with an index？

**Comment by dpvanbalen** [comment, marked as answer]
The easiest way is probably to first use `indexed` to match each element with its index, and then use `filter` on that matrix. Does this help? If not, can you elaborate a bit more on what exactly you would like this function to do?

**Reply by wujilingfeng** [reply]
> The easiest way is probably to first use `indexed` to match each element with its index, and then use `filter` on that matrix. Does this help? If not, can you elaborate a bit more on what exactly you would like this function to do?

I use the same principle as you, but the details are different
